packages/liveblocks-react-lexical/src/comments/floating-threads.tsx

```diff
@@ -0,0 +1,321 @@
+  threads: ThreadData<M>[];
+  /**
+   * Override the component's components.
+   */
+  components?: Partial<ThreadPanelComponents>;
+}
+
+export function FloatingThreads({
+  threads,
+  components,
+  ...divProps
+}: FloatingThreadsProps) {
+  const activeThreads = useActiveThreads();
+
+  const Thread = components?.Thread ?? DefaultThread;
+
+  const [editor] = useLexicalComposerContext();
+  const nodes = useThreadToNodes(); // A map of thread ids to a set of thread mark nodes associated with the thread
+
+  const [range, setRange] = useState<{
+    range: Range;
+    threads: ThreadData[];
+  } | null>(null);
+
+  const handleUpdateRange = useCallback(() => {
+    function getActiveRange(): Range | null {
+      function getActiveElements() {
+        const activeElements = new Set<HTMLElement>();
+
+        for (const thread of activeThreads) {
+          const keys = nodes.get(thread);
+          if (keys === undefined) continue;
+
+          for (const key of keys) {
+            const element = editor.getElementByKey(key);
+            if (element === null) continue;
+            activeElements.add(element);
+          }
+        }
+        return activeElements;
+      }
+
+      const activeElements = getActiveElements();
+
+      const sortedElements = Array.from(activeElements).sort(compareNodes);
+      if (sortedElements.length === 0) return null;
+
+      const range = document.createRange();
+      range.setStartBefore(sortedElements[0]);
+      range.setEndAfter(sortedElements[sortedElements.length - 1]);
+
+      return range;
+    }
+
+    const active = (threads ?? []).filter((thread) =>
+      activeThreads.includes(thread.id)
+    );
+
+    const range = getActiveRange();
+    if (range === null) {
+      setRange(null);
+      return;
+    }
+
+    setRange({ range, threads: active });
+  }, [activeThreads, nodes, editor, threads]);
+
+  useEffect(() => {
+    handleUpdateRange();
+  }, [handleUpdateRange]);
+
+  useEffect(() => {
+    return editor.registerUpdateListener(handleUpdateRange);
+  }, [editor, handleUpdateRange]);
+
+  const handleEscapeKeydown = useCallback((): boolean => {
+    if (range === null) return false;
+    setRange(null);
+    return true;
+  }, [range]);
+
+  useEffect(() => {
+    return editor.registerCommand(
+      KEY_ESCAPE_COMMAND,
+      handleEscapeKeydown,
+      COMMAND_PRIORITY_HIGH
+    );
+  }, [editor, handleEscapeKeydown]);
+
+  const isCollapsed = useIsSelectionCollapsed();
+
+  if (range === null || isCollapsed === null || !isCollapsed) return null;
+
+  return (
+    <FloatingThreadPortal
+      range={range.range}
+      container={document.body}
+      {...divProps}
+    >
+      {range.threads.map((thread) => (
+        <ThreadWrapper
+          key={thread.id}
+          thread={thread}
+          Thread={Thread}
+          onEscapeKeydown={handleEscapeKeydown}
+          className="lb-lexical-floating-threads-thread"
+        />
```

Comment:
Each `Thread` in the `FloatingThreads` component is labelled with classname `lb-lexical-floating-threads-thread`. And, because this is passed directly from the `FloatingThreads` component, users are not able to remove this classname but only add to the list of classnames. I noticed we were doing this with the old `ThreadPanel` component. Is that intentional @marcbouchenoire ?

Reply:
Yes we do this for all our "render props", it does prevent you from removing it (I think that's fine) but otherwise you'd have to hardcode it yourself to keep the default styles working, I don't think we want our users to manually set our class names on their end.